repositories/csv_user_repository.go

```diff
@@ -0,0 +1,51 @@
+package repositories
+
+import (
+	"encoding/csv"
+	"io"
+	"os"
+
+	"github.com/iszandro/golang-bootcamp-2020/models"
+)
+
+// CVSUserRepository allows to get users from a CSV file.
+type CVSUserRepository struct{}
+
+// GetUsers returns a slice of models.User by reading a CSV file.
+// It ignores first row (index 0), which has the CSV headers.
+// It loops each CSV row and builds a models.User with the CSV data.
+func (repository CVSUserRepository) GetUsers(filePath string) ([]models.User, error) {
+	file, err := os.Open(filePath)
+	defer file.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	var users []models.User
+	reader := csv.NewReader(file)
+
+	for i := 0; ; i++ {
+		record, err := reader.Read()
+
+		if i == 0 {
+			continue
+		}
+
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
```

Comment:
Imagine you are reading a big number of lines; let's say a million. With the current implementation, you should be checking for the error a million times, times two.

With the following suggested approach, you should be checking a single time for the error.

Both implementations are in the order of O(1), but hey, better checking a million times rather than two million.

```suggestion
		if err != nil {
			if err == io.EOF {
				break
			}
			return nil, err
		}
```